lib/rules/quotes.js

```diff
@@ -128,7 +128,7 @@ module.exports = {
          * @private
          */
         function isJSXElement(node) {
```

Comment:
Since this is no longer checking exclusively for `JSXElement`, perhaps rename it to `isJSX`, pass it the `Literal` node directly, and have it return `node.parent.type !== "JSXExpressionContainer"`? Then the check below becomes just "Is this `Literal` node a JSX literal or a JS literal?", and the method of checking the parent becomes just a hidden implementation detail.